tomviz/modules/ModuleManager.cxx

```diff
@@ -782,7 +794,29 @@ bool ModuleManager::deserialize(const QJsonObject& doc, const QDir& stateDir)
       proxy->UpdateVTKObjects();
     }
     viewProxy->UpdateVTKObjects();
+    if (view.contains("scaleLegend")) {
+      auto scaleLegend = ScaleLegend::getScaleLegend(viewProxy);
+      if (scaleLegend) {
+        auto legendType = view["scaleLegend"].toString();
+        if (legendType == "none") {
+          scaleLegend->setVisibility(false);
+        } else if (legendType == "cube") {
+          scaleLegend->setVisibility(true);
+          scaleLegend->setStyle(ScaleLegendStyle::Cube);
+        } else if (legendType == "ruler") {
+          scaleLegend->setVisibility(true);
+          scaleLegend->setStyle(ScaleLegendStyle::Ruler);
+        } else { // corrupted file?
+          qWarning() << "Unknown scale legend type" << legendType
+                     << "encountered in state file.  Defaulting to \"none\".";
+          scaleLegend->setVisibility(false);
+        }
+      }
+    }
   }
+  // force the view menu to update its state based on the settings we have
```

Comment:
Minor, prefer sentence with capital first letter, period at end of sentence. This could be shorter, just `// Ensure the view menu reflects restored state.` perhaps